File: ManualProg.Api/Features/Profiles/Responses/ProfileCoinTransactionResponse.cs
namespace ManualProg.Api.Features.Profiles.Responses;

public record ProfileCoinTransactionResponse
{
    public required Guid Id { get; set; }
    public required ProfileData SenderProfile { get; set; }
    public required ProfileData ReceiverProfile { get; set; }
    public required int Ammount { get; set; }
    public required DateTime Created { get; set; }

    public record ProfileData
    {
        public required Guid Id { get; set; }
        public required string Name { get; set; }
        public required bool HasImage { get; set; }
    }
}

File: ManualProg.App/src/app/profiles/data-access/responses/profile-coin-transaction-response.ts
export interface IProfileCoinTransactionResponse {
  id: string;
  senderProfile: IProfileCoinTransactionResponseProfileData;
  receiverProfile: IProfileCoinTransactionResponseProfileData;
  ammount: number;
  created: Date;
}

export interface IProfileCoinTransactionResponseProfileData {
  id: string;
  name: string;
  hasImage: boolean;
}
